src/torcontrol.cpp

```diff
@@ -414,6 +416,12 @@ TorController::TorController(struct event_base* _base, const std::string& _targe
         LogPrint(BCLog::TOR, "tor: Reading cached private key from %s\n", GetPrivateKeyFile());
         private_key = pkf.second;
     }
+    // Set Single Onion Service flag
+    if (mode == 1)
```

Comment:
Yeah could do that. Though I think the number of options of bitcoind is kind of getting out of hand, and this seems quite obscure so I don't mind hiding it a bit.